bodo/pandas/_pipeline.cpp

```diff
@@ -2,36 +2,69 @@
 
 #include "physical/result_collector.h"
 
+/*
+ * This has to be a recursive routine.  Each operator in the pipeline could
```

Comment:
I assume this is at least inspired by DuckDB. Could you add links to the corresponding DuckDB codes for reference?

Reply:
Inspired by listening to some videos of people talking about the requirements but not from looking at the code.

Reply:
Is finding the relevant code too difficult?